scripts/views/mapView.js

```diff
@@ -1,439 +1,391 @@
+    toggleFilters(['senate_glow', 'senate_fill'], filterSenate);
+    toggleFilters(['district_glow', 'district_fill'], filterDistrict);
+  }
+
+  function toggleFilters (layers, filter) {
+    layers.forEach(function (layer) {
+      map.setFilter(layer, filter);
+      map.setLayoutProperty(layer, 'visibility', 'visible');
+    })
+  }
+
+  // Refocuses the map to predetermined bounding boxes based on a state code & (optionally) a district #.
+  function focusMap (stateAbbr, districtCode) {
+    var height = window.innerHeight,
+      width = window.innerWidth,
+      districtAbbr = districtCode ? districtCode : '';
+
+    var view = geoViewport.viewport(bboxes[stateAbbr + districtAbbr], [width/2, height/2]);
+    view.zoom = view.zoom - 0.5;
+    map.flyTo(view);
+  }
+
+  // Handles the highlight for districts when clicked on. 
+  function highlightDistrict (geoid) {
+    var visibility = map.getLayoutProperty('selected-fill', 'visibility');
+    // If there are no selections, turn dem layers on
+    if (visibility === 'none') {
+      map.setLayoutProperty('selected-fill', 'visibility', 'visible');
+      map.setLayoutProperty('selected-border', 'visibility', 'visible');
+    }
+
+    // Filter for which district has been selected
+    var filter = ['all', ['==', 'GEOID', geoid]];
+
+    // Set that layer filter to the selected
+    map.setFilter('selected-fill', filter);
+    map.setFilter('selected-border', filter);
+  }
+
+  // Make the Table below the map
+  function makeTable (data) {
+    var tableRowTemplate = Handlebars.getTemplate('eventTableRow');
+    for (key in data){
+      var ele = data[key];
+      if(ele.Member){
+        var id = ele.Member+ele.Date;
+        ele.rowid = id.replace(/[\W]/g, '');
+        TownHall.allTownHalls.push(ele);
+        $('#all-events-table').append(tableRowTemplate(ele));
+      }
+    }
+  }
+
+  // Assign Zip Lookup function to Search field
+  function sidebarEvents () {
+    $('#look-up').on('submit', eventHandler.lookup);
+  }
+
+  // Zip Code Lookup!
+  eventHandler.lookup = function (e) {
+    e.preventDefault();
+    var zip = $('#look-up input').val();
+    if (zip) {
+      var validDistricts = [];
+      var callbackTrigger = 0;
+      var thisState;
+
+      zipLookup.forEach(function(n){
+        if (n.zip === zip) {
+          focusMap(n.abr, n.dis);
+          thisState = n.abr;
+          validDistricts.push(n.dis);
+        // Fall back on first four strings if exact zip match doesn't exist
+        } else if (n.zip.substring(0,4) === zip.substring(0,4)) {
+          focusMap(n.abr, n.dis);
+          thisState = n.abr;
+          validDistricts.push(n.dis);
+        }
+
+        callbackTrigger++;
+        if(callbackTrigger === zipLookup.length){
+          makeSidebar(thisState, validDistricts);
+        }
       });
-    });
+    }
   };
 
-  readData();
+  // Match the looked up zip code to district #
+  function matchSelectionToZip (state, districts) {
```

Comment:
Again, probably want to do this via an api cause districts change.  We're already doing it for the MoC cards.